File: rzops-web/src/lib/types/backup_plan.ts
export interface BackupPlanResponse {
  id: string;
  name: string;
  target_type: string | null;
  target_id: string | null;
  target_name: string | null;
  schedule: string | null;
  retention_days: number | null;
  status: string;
  remarks: string | null;
  created_at: string;
  updated_at: string;
}

export interface BackupPlanListResponse {
  data: BackupPlanResponse[];
  count: number;
}

export interface CreateBackupPlanRequest {
  name: string;
  target_type?: string;
  target_id?: string;
  schedule?: string;
  retention_days?: number;
  status?: string;
  remarks?: string;
}

export interface UpdateBackupPlanRequest extends Partial<CreateBackupPlanRequest> {}

export interface ListBackupPlansQuery {
  q?: string;
  status?: string;
  target_type?: string;
  target_id?: string;
  page?: number;
  per_page?: number;
}

File: rzops-api/domain/src/models/backup_plan.rs
use chrono::{DateTime, Utc};
use uuid::Uuid;


/// BackupPlan entity — maps to cmdb_backup_plan.
#[derive(Debug, Clone)]
pub struct BackupPlan {
    pub id: Uuid,
    pub name: String,
    pub target_type: Option<String>,
    pub target_id: Option<Uuid>,
    pub schedule: Option<String>,
    pub retention_days: Option<i32>,
    pub status: String,
    pub remarks: Option<String>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}
